code/model/DNReferenceList.php

```diff
@@ -141,13 +141,18 @@ protected function getReferences() {
 			}
 
 			if($this->getTags) {
-				$builds[$reference->getCommitHash()] = DNTag::create($reference, $this->project, $this->data);
-
+				$builds[] = DNTag::create($reference, $this->project, $this->data);
 			} else {
 				$name = $this->reference ? $this->reference->getName() : '';
-				$builds[$reference->getHash()] = DNCommit::create($reference, $this->project, $this->data, $name);
+				$builds[] = DNCommit::create($reference, $this->project, $this->data, $name);
 			}
 		}
+
+		// reverse the array so tags show with latest first.
+		if($this->getTags) {
+			return array_reverse($builds);
```

Comment:
Should commits not also be sorted in the same way if the tags are? Should be consistent I think.